### ISSUE TITLE
CDC always OOM in test case "redo_enable_downstream_failure_sync"

### ISSUE THREAD

**Issue by Tammyxia** [pingcap/tiflow#7004]
### What did you do?

- Test case: redo_enable_downstream_failure_sync , test execution url: http://rms.pingcap.net:31714/artifacts/testground/plan-exec-1231759/plan-exec-1231759-2316700259/main-logs
- 3 CDC  is always OOM, have lasted for 6h+, even worload has been stopped for several hours.
- TiDB QPS metrics:
<img width="677" alt="image" src="https://user-images.githubusercontent.com/78345569/189015813-509bd22e-89d3-4ecc-a457-beb171116f42.png">

### What did you see instead?

TiCDC metrics:
![VfJ1wCuGYc](https://user-images.githubusercontent.com/78345569/189056345-27f516a3-2e15-4a94-9921-6af4ed1879ec.jpg)

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
(paste TiCDC version here)
Release Version: v6.2.0-alpha
Git Commit Hash: 6fba81295d9f6308f044d8a0ece3ae4624bd124a
Git Branch: heads/refs/tags/v6.2.0-alpha
UTC Build Time: 2022-09-07 11:02:00
Go Version: go version go1.18.5 linux/amd64
Failpoint Build: false
```

**Comment by nongfushanquan** [comment]
I double this is caused by the table quota, since the downstream failure  won't  release the events stored in the sinker .

**Comment by asddongmen** [comment]
It only occurs in v6.2.0 which is not a LTS version, and it has never occured in v6.3.0 yet, so I decide to adjust its severity to minor and we'll do more test in v6.3.0. If v6.3.0 can always pass this case, we'll close this issue.

**Comment by asddongmen** [comment]
Confirm that it is a sinkV2 related bug, and it has been fixed in v6.3.0 and it does not exist in v6.2.0.
